### PR TITLE
ansible_galaxy_install: use locale C tentatively, else en_US

### LINKED ISSUE #5661: Calling ansible_galaxy_install generates "unsupported locale setting" error
### Summary

Installing a module with `ansible-galaxy` works no problem, but when trying to install a collection with `community.general.ansible_galaxy_install` if generates the following error:
```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "cmd": "/home/matt/.local/bin/ansible-galaxy --version",
    "invocation": {
        "module_args": {
            "ack_ansible29": false,
            "ack_min_ansiblecore211": false,
            "dest": null,
            "force": true,
            "name": "kubernetes.core",
            "no_deps": false,
            "requirements_file": null,
            "type": "collection"
        }
    },
    "msg": "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting",
    "rc": 1,
    "stderr": "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting\n",
    "stderr_lines": [
        "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting"
    ],
    "stdout": "",
    "stdout_lines": []
}
```

### Issue Type

Bug Report

### Component Name

ansible_galaxy_install

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.14.0]
  config file = None
  configured module search path = ['/home/matt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/matt/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/matt/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/matt/.local/bin/ansible
  python version = 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /home/matt/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 6.0.1  

# /home/matt/.local/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 6.0.1 
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
CONFIG_FILE() = None
```

### OS / Environment

Ubuntu 22.04

### Steps to Reproduce

```yaml role task
- name: Install Ansible collection kubernetes.core
  community.general.ansible_galaxy_install:
    type: collection
    name: kubernetes.core
```

Locale is legal for the UK.
```sh
$ locale
LANG=en_GB.UTF-8
LANGUAGE=
LC_CTYPE="en_GB.UTF-8"
LC_NUMERIC="en_GB.UTF-8"
LC_TIME="en_GB.UTF-8"
LC_COLLATE="en_GB.UTF-8"
LC_MONETARY="en_GB.UTF-8"
LC_MESSAGES="en_GB.UTF-8"
LC_PAPER="en_GB.UTF-8"
LC_NAME="en_GB.UTF-8"
LC_ADDRESS="en_GB.UTF-8"
LC_TELEPHONE="en_GB.UTF-8"
LC_MEASUREMENT="en_GB.UTF-8"
LC_IDENTIFICATION="en_GB.UTF-8"
LC_ALL=en_GB.UTF-8
```

### Expected Results

It should just install or skip the module if already installed.

Installing via the command line works no problem

```sh
$ ansible-galaxy collection install kubernetes.core --force
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Downloading https://galaxy.ansible.com/download/kubernetes-core-2.3.2.tar.gz to /home/matt/.ansible/tmp/ansible-local-54212cgr1gdmr/tmpkklx4jfb/kubernetes-core-2.3.2-351kys1s
Installing 'kubernetes.core:2.3.2' to '/home/matt/.ansible/collections/ansible_collections/kubernetes/core'
kubernetes.core:2.3.2 was installed successfully
```

`ansible-galaxy --version` also returns OK
```sh
$ /home/matt/.local/bin/ansible-galaxy --version
ansible-galaxy [core 2.14.0]
  config file = None
  configured module search path = ['/home/matt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/matt/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/matt/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/matt/.local/bin/ansible-galaxy
  python version = 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

### Actual Results

```console (paste below)
fatal: [localhost]: FAILED! => {
    "changed": false,
    "cmd": "/home/matt/.local/bin/ansible-galaxy --version",
    "invocation": {
        "module_args": {
            "ack_ansible29": false,
            "ack_min_ansiblecore211": false,
            "dest": null,
            "force": true,
            "name": "kubernetes.core",
            "no_deps": false,
            "requirements_file": null,
            "type": "collection"
        }
    },
    "msg": "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting",
    "rc": 1,
    "stderr": "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting\n",
    "stderr_lines": [
        "ERROR: Ansible could not initialize the preferred locale: unsupported locale setting"
    ],
    "stdout": "",
    "stdout_lines": []
}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tries to use locale `C.UTF-8`, if that fails then fall back to `en_US.UTF-8`.

Fixes #5661 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ansible_galaxy_install

### CODE REVIEW

**Comment by russoz** [comment]
@felixfontein I must be doing something dumb again, but I cannot find why that one test is failing. It does not even seem to be related to the change made.

**Comment by russoz** [comment]
Actually, I think I found it. The invalid locale stderr output seems to be different in that test platform.

File: plugins/modules/ansible_galaxy_install.py [hunk 1]
```diff
@@ -239,12 +249,18 @@ def process(rc, out, err):
...
-        with self.runner("version", check_rc=True, output_process=process) as ctx:
-            return ctx.run(version=True)
+        try:
+            runner = self._make_runner("C.UTF-8")
+            with runner("version", check_rc=False, output_process=process) as ctx:
+                return runner, ctx.run(version=True)
+        except UnsupportedLocale as e:
+            runner = self._make_runner("en_US.UTF-8")
+            with runner("version", check_rc=True, output_process=process) as ctx:
+                return runner, ctx.run(version=True)
```
Comment (felixfontein) [inline plugins/modules/ansible_galaxy_install.py:259]: What will happen when the `UnsupportedLocale` exception is raised here another time? Will the module just crash, or how will it react?

**Review by felixfontein — COMMENTED** [review]
@russoz great news, thanks for fixing this! It needs a changelog fragment though :)

File: plugins/modules/ansible_galaxy_install.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/ansible_galaxy_install.py:259]: Being a ModuleHelper, any exception will be captured and `module.fail_json()` accordingly.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!
@laidbackware thanks for testing!

**Pull request merged**